cypress/plugins/nodeManager/plugin.ts

```diff
@@ -0,0 +1,294 @@
+      }
+    );
+    const json = await identitiesResponse.json();
+
+    this.peerAddress = `${json.peerId}@${this.host}:${this.peerPort}`;
+
+    this.logger.log("node onboarded successfully");
+  }
+
+  async stop(): Promise<void> {
+    if (!this.process) {
+      throw "Tried to stop node before it was started";
+    }
+
+    this.process.kill();
+  }
+}
+
+class NodeManager {
+  private managedNodes: Node[] = [];
+  private logger: Logger;
+
+  constructor() {
+    this.logger = new Logger({ prefix: `[nodeManager] ` });
+  }
+
+  private getNode = (id: NodeId) => {
+    const node = this.managedNodes.find(node => {
+      return node.id === id;
+    });
+
+    if (!node) {
+      throw `Could not find node by id ${id}`;
+    }
+
+    return node;
+  };
+
+  async startNode(options: StartNodeOptions): Promise<void> {
+    this.logger.log("startNode");
+
+    const node = new Node(options);
+    await node.start();
+    this.managedNodes.push(node);
+  }
+
+  async onboardNode(options: OnboardNodeOptions): Promise<void> {
+    this.logger.log("onboardNode");
+
+    const node = this.getNode(options.id);
+
+    // FIXME
+    await node.onboard(options as NodeOnboardOptions);
+  }
+
+  async connectNodes(options: ConnectNodesOptions) {
+    this.logger.log("connectNodes");
+
+    if (options.nodeIds.length < 2) {
+      throw "Supply at least 2 node IDs";
+    }
+
+    const firstNode = this.getNode(options.nodeIds[0]);
+    const remainingNodes = this.managedNodes.filter(node => {
+      return firstNode.id !== node.id;
+    });
+
+    await fetch(
+      `http://${firstNode.host}:${firstNode.httpPort}/v1/session/settings`,
+      {
+        method: "post",
+        body: JSON.stringify({
+          appearance: { theme: "dark", hints: { showRemoteHelper: true } },
+          coco: {
+            seeds: remainingNodes.map(node => node.peerAddress),
+          },
+        }),
+        headers: {
+          Cookie: `auth-token=${firstNode.authToken}`,
+          "Content-Type": "application/json",
+        },
+      }
+    );
+
+    return null;
+  }
+
+  async getOnboardedNodes(): Promise<OnboardedNode[]> {
+    this.logger.log("getOnboardedNodes");
+
+    const onboardedNodes: OnboardedNode[] = [];
+
+    this.managedNodes.forEach(node => {
+      if (node.authToken && node.peerAddress && node.httpPort) {
+        onboardedNodes.push({
+          authToken: node.authToken,
+          peerAddress: node.peerAddress,
+          httpPort: node.httpPort,
+        });
+      }
+    });
+
+    return onboardedNodes;
+  }
+
+  async stopAllNodes(): Promise<void> {
+    this.logger.log("stopAllNodes");
+
+    this.managedNodes.forEach(async node => {
+      await node.stop();
+    });
+
+    this.managedNodes = [];
+  }
+}
+
+const nodeManager = new NodeManager();
+
+export const nodeManagerPlugin = {
+  [Commands.StartNode]: async (options: StartNodeOptions): Promise<null> => {
+    await nodeManager.startNode(options);
+
+    return null;
+  },
```

Comment:
Is there a particular reason we return `null` instead of not returning anything and using `Promise<void>` as the return type?

Reply:
Yes, it's documented in the Cypress [task API](https://docs.cypress.io/api/commands/task.html#Usage):

> The task plugin event handler can return a value or a promise. The command will fail if undefined is returned or if the promise is resolved with undefined. This helps catch typos or cases where the task event is not handled.
>
> If you do not need to return a value, **explicitly return null to signal that the given event has been handled**.